src/plugins/data/common/index_patterns/types.ts

```diff
@@ -18,6 +18,9 @@ export type FieldFormatMap = Record<string, SerializedFieldFormat>;
 
 export type RuntimeType = typeof RUNTIME_FIELD_TYPES[number];
 export interface RuntimeField {
+  /**
+   * Also used as a IndexPatternField's ES type
```

Comment:
Suggestion:

Users can override a field's ES field types (`FieldSpec.esTypes`) by specifying a RuntimeField with a type. Elasticsearch only supports a subset of field types for RuntimeFields.